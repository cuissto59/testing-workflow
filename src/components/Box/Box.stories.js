import Box from './Box';

export default {
  title: 'Layout/Box',
  component: Box,
};

const Template = (args) => <Box {...args} />;

export const Default = Template.bind({});
Default.args = {
  className: '',
  children: 'Box text',
};
