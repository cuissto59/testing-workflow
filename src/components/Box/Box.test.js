import { render } from "@testing-library/react";
import Box from "./Box";

describe("Box test", () => {
  it("should match the snapshot when children and className not given", () => {
    const { container } = render(<Box />);
    expect(container).toMatchSnapshot();
  });

  it("should match the snapshot when children given", () => {
    const children = "Box";
    const { container } = render(<Box>{children}</Box>);
    expect(container).toMatchSnapshot();
  });

  it("should match the snapshot when className given", () => {
    const className = "text-red-500";
    const { container } = render(<Box className={className} />);
    expect(container).toMatchSnapshot();
  });

  it("should match the snapshot when children and className given", () => {
    const className = "text-red-500";
    const children = "Box";
    const { container } = render(<Box className={className}>{children}</Box>);
    expect(container).toMatchSnapshot();
  });
});
