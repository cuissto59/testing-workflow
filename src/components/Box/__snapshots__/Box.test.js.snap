// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box test should match the snapshot when children and className given 1`] = `
<div>
  <div
    class="text-red-500"
  >
    Box
  </div>
</div>
`;

exports[`Box test should match the snapshot when children and className not given 1`] = `
<div>
  <div />
</div>
`;

exports[`Box test should match the snapshot when children given 1`] = `
<div>
  <div>
    Box
  </div>
</div>
`;

exports[`Box test should match the snapshot when className given 1`] = `
<div>
  <div
    class="text-red-500"
  />
</div>
`;
